internal/provider/transfer_resource.go

```diff
@@ -335,3 +302,578 @@ func transferTypeValidator() validator.String {
+	var diags diag.Diagnostics = nil
+
+	m.Id = types.StringValue(t.GetId())
+	m.ProjectID = types.StringValue(t.GetProjectId())
+	m.Name = types.StringValue(t.GetName())
+	m.Description = types.StringValue(t.GetDescription())
+	m.Source = types.StringValue(t.GetSource().GetId())
+	m.Target = types.StringValue(t.GetTarget().GetId())
+	m.Type = types.StringValue(t.GetType().String())
+
+	if t.GetTransformation() != nil {
+		if m.Transformation == nil {
+			m.Transformation = new(transferTransformation)
+		}
+		diags.Append(m.Transformation.Read(t.Transformation)...)
+	} else {
+		m.Transformation = nil
+	}
+
+	return nil
+}
+
+func (m *transferResourceModel) UpdateRequest() (*transfer.UpdateTransferRequest, diag.Diagnostics) {
+	r := new(transfer.UpdateTransferRequest)
+	var diags diag.Diagnostics = nil
+	var dgs diag.Diagnostics = nil
+
+	r.TransferId = m.Id.ValueString()
+	r.Name = m.Name.ValueString()
+	r.Description = m.Description.ValueString()
+
+	if m.Transformation != nil {
+		r.Transformation, dgs = m.Transformation.AsRequest(requestTypeUpdate)
+		diags.Append(dgs...)
+	}
+
+	return r, diags
+}
+
+func (m *transferResourceModel) DeleteRequest() (*transfer.DeleteTransferRequest, diag.Diagnostics) {
+	return &transfer.DeleteTransferRequest{
+		TransferId: m.Id.ValueString(),
+	}, nil
+}
+
+type transferTransformation struct {
+	Transformers []transferTransformer `tfsdk:"transformers"`
+}
+
+func transferTransformationSchema() schema.Attribute {
+	return schema.SingleNestedAttribute{
+		Attributes: map[string]schema.Attribute{
+			"transformers": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"replace_primary_key": transferTransformerReplacePrimaryKeySchema(),
+						"convert_to_string":   transferTransformerConvertToStringSchema(),
+						"dbt":                 transferTransformerDBTSchema(),
+						"table_splitter":      transferTransformerTableSplitterSchema(),
+					},
+				},
+				Optional: true,
+			},
+		},
+		Optional: true,
+	}
+}
+
+func (m *transferTransformation) AsRequest(rqt requestType) (*transfer.Transformation, diag.Diagnostics) {
+	r := new(transfer.Transformation)
+	var diags diag.Diagnostics = nil
+	var dgs diag.Diagnostics = nil
+
+	if len(m.Transformers) > 0 {
+		r.Transformers = make([]*transfer.Transformer, len(m.Transformers))
+		for i := range m.Transformers {
+			r.Transformers[i], dgs = m.Transformers[i].AsRequest(rqt)
+			diags.Append(dgs...)
+		}
+	}
+
+	return r, diags
+}
+
+func (m *transferTransformation) Read(t *transfer.Transformation) diag.Diagnostics {
+	var diags diag.Diagnostics = nil
+
+	tTransformers := t.GetTransformers()
+	if len(tTransformers) > 0 {
+		if len(m.Transformers) == 0 {
+			m.Transformers = make([]transferTransformer, 0)
+		}
+		for i := range tTransformers {
+			if i >= len(m.Transformers) {
+				m.Transformers = append(m.Transformers, *new(transferTransformer))
```

Comment:
What do you mean? `transferTransformer` contains a set of fields different from a set of fields in an object carrying Protobuf fields.

Instead of making copies, this code re-`Read`s existing transformers.

Reply:
My bad, I misread it. 
Everything is fine here.

I would just write it in a different manner:

```go
t := &transferTransformer{}
diags.Append(t.Read(Transformers[i])...)
append(m.Transformers, t)
```

Reply:
Yes, I also changed my mind about the singature of `Read()`. This one is better, as it requires no intermediate object.